### PR TITLE
PXP-613: Update API calls order

### PR DESCRIPTION
**Pull request by mfauzaan**
## Summary
API endpoints are being called on the TUI without proper verification of the namespace and version. 

Fixed by adjusting the order of API calls to ensure that namespace and version validation is performed before making API requests.

Ticket: https://mindvalley.atlassian.net/browse/PXP-613

## What's Changed

- [x] Make the current_namespace and version types optional (Optional<String>) to allow for cases where they may not have a value.
- [x] Break down the network_event into individual functions to make the code easier to understand and maintain.
- [x] Ensure that the names of network events accurately reflect the underlying logic and operations.
- [x] Modify the code responsible for selecting namespaces and versions to work with the updated data type.

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/147#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by onimsha — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/147#" title="Author self-approved">mfauzaan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/147#pullrequestreview-1632453998" title="Approved">onimsha</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
